lib/govuk_app_config/govuk_content_security_policy.rb

```diff
@@ -11,7 +11,11 @@ module GovukContentSecurityPolicy
   # - https://csp-evaluator.withgoogle.com
   # - https://cspvalidator.org
 
-  GOVUK_DOMAINS = "'self' *.publishing.service.gov.uk localhost".freeze
+  GOVUK_DOMAINS = [
+    "'self'",
+    '*.publishing.service.gov.uk',
+    "*.#{ENV['GOVUK_APP_DOMAIN'] || ENV['DEV_DOMAIN'] || 'dev.gov.uk'}"
```

Comment:
Nice idea 👍 

I know this will probably make you sad but we should probably start with GOVUK_APP_DOMAIN_EXTERNAL before GOVUK_APP_DOMAIN - since otherwise this may end up with the .digital internal AWS domains. 

Have you seen DEV_DOMAIN used outside plek at all? Since it's a non GOVUK env var I wonder if we can drop it here - feels like you should just use GOVUK_APP_DOMAIN for it anyway.

Reply:
Good point :-). I kind of wish we'd lead with 'DOMAIN_INTERNAL' on the other one, since that's what they sometimes are. Anyway, I've updated as suggested, since I agree 'DEV_DOMAIN' should use an official prefix if it wants to be supported.

Reply:
I agree about domain internal, it's a pain point. Sadly you'll have to also check GOVUK_APP_DOMAIN after GOVUK_APP_DOMAIN_EXTERNAL as the external one is not always set :(

Have you been able to test government-frontend to check it's tests still pass with these changes? 

Nice one with regards to Plek 👍

Reply:
😭

Reply:
Forgot to mention - I've checked government-frontend after each change. All I can say is that it works for dev.gov.uk, and the fact that we still hardcode publishing... means the functionality won't change otherwise.